pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -683,7 +684,7 @@ func (alias *Alias) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
 		return err
 	}
 	*alias = Alias(userAlias)
-	alias.Name = alias.Name[len(UserAliasPrefix):]
+	alias.Name = strings.TrimPrefix(alias.Name, UserAliasPrefix)
```

Comment:
I think that `TrimPrefix` kind of does the same, no? 
https://golang.org/src/strings/strings.go?s=22858:22898#L892

Reply:
Following the same link, https://golang.org/src/strings/strings.go?s=22858:22898#L892, `TrimPrefix` does: `// If s doesn't start with prefix, s is returned unchanged.` 

The reason for this PR was that we have seen that some devices' alias names got evolved into things like `ua-let0` (from `tablet0`), `ua-tio` (from `virtio`) in downstream tests involving Windows VM migration and shutdown. So the speculated lifecycle of these alias names, `tablet0` as an example, would likely be: 
      ```tablet0 --UnmarshalXML--> let0 --MarshalXML--> ua-let0``` 

For the said bug scenario, we could not quite figure out why the "unmarshalled" alias name, e.g. `tablet0`, was originated. But this fix will prevent such short "unmarshalled" alias name got trimmed shorter, leading to multiple devices having the same alias name.

Reply:
Sure. I'm not against this change :) I'm just asking to update the description.

Reply:
I would suggest that we fix it completely. The current change only fixes it for the first unmarshalling. Subsequent marshallings will then serialize it then without the `ua-` prefix, no matter if it had one before or not.

Can we do the following:

 1) the current `UserAlias` struct makes no sense at all, it is only used in the marshal and unmarshal functions and can as well be removed completely.
 2) If we really want to hide this `ua-` prefix handling correclty, I think we should make the `Name`  attribute private in the struct, add a `userDefined` boolean to the struct and then:
 3) Add an explicit constructors: `func UserDefinedAlias(name string) Alias` and make use of it in converter.go
 4) In the Marshal and Unmarshal overloads for XML, we can then properly detect the prefix , trim the prefix and set `userDefined` properly and vice-versa.

Reply:
@rmohr, the key of your layout appears to rely on the knowledge that uses of the explicit constructor for `UserAlias` object are contained only in `converter.go`. I will go ahead and code in this layout.

Reply:
> @rmohr, the key of your layout appears to rely on the knowledge that uses of the explicit constructor for `UserAlias` object are contained only in `converter.go`. I will go ahead and code in this layout.

That's exactly it. Perfect :+1: